Cockpit 242 (#399)
cockpit-project/cockpit-project.github.io: _posts/2021-04-14-cockpit-242.md

---
title: Cockpit 242
author: marusak
date: 2021-04-14
tags: cockpit linux
slug: cockpit-242
category: release
summary: Cockpit with split out Machines page and website refresh
---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.

Here are the release notes from Cockpit version 242.

## Groundwork for Snowpack support

> [Snowpack](https://www.snowpack.dev/) is a web project build system alternative to [webpack](https://webpack.js.org/). Previously, when using it to build a Cockpit page which used an [npm module](https://www.npmjs.com/) containing an `@`, such as [@patternfly](https://www.npmjs.com/package/@patternfly/react-core), the page [could not be loaded](https://github.com/cockpit-project/cockpit/issues/15288). This has been fixed.

Reviewer: ```suggestion
[Snowpack](https://www.snowpack.dev/) is a web project build system alternative to [webpack](https://webpack.js.org/). Previously, when using Snowpack to build Cockpit pages using an [npm module](https://www.npmjs.com/) with an `@`, the page [could not be loaded](https://github.com/cockpit-project/cockpit/issues/15288). This has been fixed. As Cockpit uses [@patternfly](https://www.npmjs.com/package/@patternfly/react-core) in this style, it was an important first step.
```

This is really wordy overall and I lost what "it" was referring to. (I _know_ what is meant, but because there are so many different things it could refer to, and the last thing mentioned is webpack, I decided being specific in this case is best.)

Additionally, I couldn't find a great way to break up the sentence and keep the exact same flow. So I broke it up and changed the order of things a bit.

> Note that there is not yet a fully working and supported example of a snowpack-built Cockpit project, just some [initial experimentation](https://piware.de/post/2021-02-05-snowpack/).

Reviewer: ```suggestion
Please be aware that there is not a fully working and supported example of a Snowpack-built Cockpit project at this time, just some [initial experimentation](https://piware.de/post/2021-02-05-snowpack/).
```

- Changed sentence intro (optional)
- Fixed case of Snowpack
- Replaced _yet_ with _at this time_ (in a different part of the sentence)

## Machines: Split out to separate project

> The Machines page is by far the largest Cockpit component. It has now been split out into the standalone [cockpit-machines project](https://github.com/cockpit-project/cockpit-machines/) to make it easier and faster to develop. The same split has also happened in [Fedora](https://bodhi.fedoraproject.org/updates/?packages=cockpit-machines) and [Debian](https://tracker.debian.org/pkg/cockpit-machines).

Reviewer: ```suggestion
The Machines page is easily the largest Cockpit component. It has been split out to the standalone [cockpit-machines project](https://github.com/cockpit-project/cockpit-machines/), making it easier and faster to develop. This same split has also happened in [Fedora](https://bodhi.fedoraproject.org/updates/?packages=cockpit-machines) and [Debian](https://tracker.debian.org/pkg/cockpit-machines) packaging.
```

- Replaced "by far" idiom
- Removed unnecessary "now"
- Misc. changes
- Specified packaging

## Cockpit website refresh

[Cockpit website](https://cockpit-project.org/) has been restyled and refreshed. This update includes:
- a fresh style
- completely new content on the front page
- updated screenshots

> - a brand new docs page

Reviewer: ```suggestion
- a brand new [docs page](https://cockpit-project.org/documentation.html)
```

Added URL

---
title: Cockpit 242
author: marusak
date: 2021-04-14
tags: cockpit linux
slug: cockpit-242
category: release

> summary: Cockpit ...

Reviewer 2: .. with split out Machines page and website refresh?

---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.

Here are the release notes from Cockpit version 242.

> ## Support for pages built with snowpack

Reviewer: Perhaps one of these instead:

```suggestion
## Initial Snowpack support for pages
```

```suggestion
## Groundwork for Snowpack support
```

Reviewer: Snowpack is a proper name of a project, so it should be capitalized, unless the project decides to invoke custom stylistic rules. (Note: They haven't.)

Reviewer: Of course, if you're referring to the command `snowpack`, that's different. But for the headline, it should be talking about Snowpack as a whole, not the specific command to invoke it.

Reviewer 2: I did mean the project name, so capitalizing it sounds appropriate. Its website uses "Snowpack".

> [Snowpack](https://www.snowpack.dev/) is a web project build system alternative to [webpack](https://webpack.js.org/). When using it to build a Cockpit page that uses an [npm module](https://www.npmjs.com/) containing a `@`, such as [@patternfly](https://www.npmjs.com/package/@patternfly/react-core), the page [could not be loaded](https://github.com/cockpit-project/cockpit/issues/15288). This has been fixed now. Note that there is not yet a fully working and supported example of a snowpack-built Cockpit project, just some [initial experimentation](https://piware.de/post/2021-02-05-snowpack/).

Reviewer 3: ```suggestion
[Snowpack](https://www.snowpack.dev/) is a web project build system alternative to [webpack](https://webpack.js.org/). Previously, when using it to build a Cockpit page which used an [npm module](https://www.npmjs.com/) containing an `@`, such as [@patternfly](https://www.npmjs.com/package/@patternfly/react-core), the page [could not be loaded](https://github.com/cockpit-project/cockpit/issues/15288). This has been fixed.

Note that there is not yet a fully working and supported example of a snowpack-built Cockpit project, just some [initial experimentation](https://piware.de/post/2021-02-05-snowpack/).
```

## Machines: Split out to separate project

> The Machines page is by far the largest Cockpit component. It got split out to the standalone [cockpit-machines project](https://github.com/cockpit-project/cockpit-machines/) to make it easier and faster to develop. The same split happened in [Fedora](https://bodhi.fedoraproject.org/updates/?packages=cockpit-machines) and [Debian](https://tracker.debian.org/pkg/cockpit-machines).

Reviewer 3: ```suggestion
The Machines page is by far the largest Cockpit component. It has now been split out into the standalone [cockpit-machines project](https://github.com/cockpit-project/cockpit-machines/) to make it easier and faster to develop. The same split has also happened in [Fedora](https://bodhi.fedoraproject.org/updates/?packages=cockpit-machines) and [Debian](https://tracker.debian.org/pkg/cockpit-machines).
```

> ## Cockpit website refreshed

Reviewer: ```suggestion
## Cockpit website refresh
```

Changed "refreshed" from a past-tense verb to the noun "refresh".

> It includes:

Reviewer 3: You refer to an "it" which has so far only been introduced in the heading, which is not our usual style.  Add a short text about how we have a new website.

- a fresh style
- completely new content on the front page
- updated screenshots
- a brand new docs page

Check it out!

> https://cockpit-project.org/

Reviewer 2: That should really be a link, aside from the fact that most people will look at that from that very site :grin: (except from RSS readers, so it still has merits)

Author: Removed it and added it at beginning of paragraph as link.
Fair point that majority of people will see this blog once they navigate from the website, but still a nice one to mention :)

> ## Cockpit ecosystem updates

Reviewer: This is more fun:

"Updates across the Cockpitverse"

...but perhaps _too_ fun. :laughing:

Author: I think that many people would struggle to parse it as `Cockpit ecosystem/universe`

Reviewer: Yep, too fun. :wink: That's why I didn't officially suggest it.

> We also today released new versions of [cockpit-podman](https://github.com/cockpit-project/cockpit-podman/), [cockpit-ostree](https://github.com/cockpit-project/cockpit-ostree) and [cockpit-machines](https://github.com/cockpit-project/cockpit-machines/).

Reviewer 3: ```suggestion
Today, we're also releasing new versions of [cockpit-podman](https://github.com/cockpit-project/cockpit-podman/), [cockpit-ostree](https://github.com/cockpit-project/cockpit-ostree), and the newly-independent [cockpit-machines](https://github.com/cockpit-project/cockpit-machines/).
```

> We also today released new versions of [cockpit-podman](https://github.com/cockpit-project/cockpit-podman/), [cockpit-ostree](https://github.com/cockpit-project/cockpit-ostree) and [cockpit-machines](https://github.com/cockpit-project/cockpit-machines/).

Reviewer 2: "Today we also..", or "We also released... today." (time goes to the beginning or end). Can you link directly to the corresponding release/version page?
